repos/system_upgrade/el7toel8/actors/networkmanagerupdateservice/actor.py

```diff
@@ -0,0 +1,56 @@
+from leapp.actors import Actor
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+
+import os
+import six
+import subprocess
+from subprocess import CalledProcessError
+
+# On RHEL7 if the NetworkManager service was disabled and
+# NetworkManager-wait-online enabled, the former would not be
+# started. This changed on RHEL8, where NM-w-o 'Requires' NM and so NM
+# can be started even if disabled. Upon upgrade, to keep the previous
+# behavior we must disable NM-w-o when NM is disabled.
+# See also: https://bugzilla.redhat.com/show_bug.cgi?id=1520865
+
+class NetworkManagerUpdateService(Actor):
+    name = 'network_manager_update_service'
+    description = 'This actor updates NetworkManager services status when needed.'
+    consumes = ()
+    produces = ()
+    tags = (ApplicationsPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        nm_enabled = self.unit_enabled('NetworkManager.service')
+        nmwo_enabled = self.unit_enabled('NetworkManager-wait-online.service')
+        self.log_services_state('initial', nm_enabled, nmwo_enabled)
+
+        if not nm_enabled and nmwo_enabled:
+            self.log.info('Disabling NetworkManager-wait-online.service')
+            self.call(['systemctl', 'disable', 'NetworkManager-wait-online.service'])
```

Comment:
This might raise, therefore please guard this from escaping - Non caught exceptions are considered a bug. There are only certain allowed exceptions like StopActorExecution and StopActorExecutionError which will gracefully be handled by the framework and handled accordingly.